segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/cache/CacheLayout.java

```diff
@@ -0,0 +1,351 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.cache;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+/**
+ * Base class that defines the memory layout for a block-based {@link CacheStorage}.
+ */
+abstract class CacheLayout {
+    //region Members
+
+    /**
+     * The maximum number of bytes that can be stored in a {@link CacheStorage} using this type of layout.
+     */
+    static final long MAX_TOTAL_SIZE = 256 * 1024 * 1024 * 1024L;
```

Comment:
The reason I chose this "small" max value is because of the number of bits available in the `address` of each entry. There is room for more data in that 32-bit address, but at the moment I didn't have a need for all of them. For Pravega's current needs 256GB (per process) is good enough. If we run in a situation where we need more, we can always change this to support future needs. (there is no need to be too preemptive here as this is not persistent data so we need not worry about any versioning, etc.).